Subject: [PATCH] m68k: macmace doesn't check for Mac

From: Geert Uytterhoeven <geert@linux-m68k.org>

The Macintosh MACE network driver (macmace) doesn't check whether it's
actually running on Mac hardware, causing a crash if it isn't.

Signed-off-by: Geert Uytterhoeven <geert@linux-m68k.org>
---
 drivers/net/macmace.c |    3 +++
 1 file changed, 3 insertions(+)

--- a/drivers/net/macmace.c
+++ b/drivers/net/macmace.c
@@ -781,6 +781,9 @@ static int __init mac_mace_init_module(v
 {
 	int err;
 
+	if (!MACH_IS_MAC)
+		return -ENODEV;
+
 	if ((err = platform_driver_register(&mac_mace_driver))) {
 		printk(KERN_ERR "Driver registration failed\n");
 		return err;
